core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/HttpAuthOption.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi;
+
+import java.util.function.BiConsumer;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.auth.spi.internal.DefaultHttpAuthOption;
+import software.amazon.awssdk.identity.spi.IdentityProperty;
+import software.amazon.awssdk.utils.builder.SdkBuilder;
+
+/**
+ * An authentication scheme option, composed of the scheme ID and properties for use when resolving the identity and signing
+ * the request.
+ *
+ * This is used in the output from the auth scheme resolver. The resolver returns a list of these, in the order the auth scheme
+ * resolver wishes to use them.
+ *
+ * @see HttpAuthScheme
+ */
+@SdkProtectedApi
```

Comment:
I can no longer remember all discussions in my head.. HttpAuthOption was intended for our internal resolver logic, but I'm not sure either if it can be. Seems we can punt this decision though, if we keep it protected for now and see what the requirements/SEP comes up with.

Reply:
yeah, sounds good. always good to start stricter and open up when needed.